src/libstd/lib.rs

```diff
@@ -469,6 +469,7 @@ pub mod error;
 pub mod ffi;
 pub mod fs;
 pub mod io;
+#[cfg(not(target_os = "l4re"))]
```

Comment:
I see your point and I know that it is unusual. As Sebastian already mentioned "L4Re looks like a normal (uClibc) Unix, just without networking". Implementing a completely new platform would lead to large amounts of redundant code while we are able to (even if it is unusual) can completely disable the net part with three statements.